telegram/dcs/resolver.go

```diff
@@ -93,23 +98,25 @@ func (p plain) connect(ctx context.Context, dc int, dcOptions []tg.DCOption) (tr
 
 // PlainOptions is plain resolver creation options.
 type PlainOptions struct {
-	// Transport to use.
-	Transport Transport
-	// Dialer to use. Default net.Dialer will be used by default.
-	Dialer transport.Dialer
-	// Network to use.
+	// Protocol is the transport protocol to use. Defaults to intermediate.
+	Protocol Protocol
+	// DialContext specifies the dial function for creating unencrypted TCP connections.
+	// If DialContext is nil, then the resolver dials using package net.
+	DialContext DialFunc
```

Comment:
Please use Dial instead of DialContext here and in other places 

The Context postfix is redundant as we don't have Dial without context.

Reply:
You are right, also it’s probably a good idea to default to `proxy.Dial` since that’s what we use in `telegram` package.